exercises/stack_01.js

```diff
@@ -0,0 +1,82 @@
+window.Collections = {};
+
+(function(module) {
+    function Stack(arr) {
+        /* constructor */
+        this.arr = arr;
+    };
+
+    /* creating a forEach method on Stack */
+    Stack.prototype.forEach = function(callback) {
+        for (var i = 0; i < this.arr.length; i++) {
+            callback(this.arr[i], i, this.arr);
+        }
+    };
+
+    Stack.prototype.push = function(elem){
+        // this.arr.push(elem);
+        this.arr[this.arr.length] = elem;
+    }
+
+    Stack.prototype.peek = function() {
+        return this.arr[this.arr.length-1];
+    }
+
+    Stack.prototype.pop = function() {
+        // return this.arr.pop();
+        var i = this.arr.length-1,
+            val = this.arr[i];
+        this.arr.splice(i, 1);
+        return val;
```

Comment:
Commented out version is fine, too. But, this is good!

Reply:
same as above